torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -15,8 +17,8 @@
 namespace torch {
 namespace jit {
 
-thread_local bool profiling_mode = false;
-bool& getProfilingMode() {
+static std::atomic<bool> profiling_mode{false};
+std::atomic<bool>& getProfilingMode() {
```

Comment:
atomic or thread local?

Reply:
atomic/global, otherwise, we might miss the graphs that we should be testing with `ProfilingGraphExecutor` because autograd just schedules them on another thread.